src/policy/space.rs

```diff
@@ -355,10 +355,11 @@ pub trait Space<VM: VMBinding>: 'static + SFT + Sync + Downcast {
             bytes,
             new_chunk
         );
-        debug_assert!(
-            (new_chunk && start.is_aligned_to(BYTES_IN_CHUNK)) || !new_chunk,
-            "should only grow space for new chunks at chunk-aligned start address"
-        );
+        // FIXME: This assertion is too strict. See https://github.com/mmtk/mmtk-core/issues/374
+        // debug_assert!(
```

Comment:
I am not comfortable about just removing this assertion. As discussed in https://github.com/mmtk/mmtk-core/issues/374#issuecomment-878047774, if somehow, we have an address in the middle of a chunk and tell the function that this is a new chunk, the obvious problem is that this is NOT a new chunk and we must have already initialised the SFT map for the chunk.

By the way, why is this change included in this PR? This PR changes GC, I assume that wouldn't trigger this assertion.

Reply:
When I test this PR with the new DaCapo benchmarks, it sometimes triggers this assertion failure. I feel that this assertion is too strict, which means this is actually an incorrect assertion. We shouldn't put an incorrect assertion in the code base.